src/main/java/org/fxmisc/flowless/VirtualizedScrollPane.java

```diff
@@ -0,0 +1,203 @@
+    private final V content;
+
+    private Var<Double> hbarValue;
+    private Var<Double> vbarValue;
+
+    public VirtualizedScrollPane(V content) {
+        this.getStyleClass().add("virtualized-scroll-pane");
+        this.content = content;
+
+        // create scrollbars
+        hbar = new ScrollBar();
+        vbar = new ScrollBar();
+        hbar.setOrientation(Orientation.HORIZONTAL);
+        vbar.setOrientation(Orientation.VERTICAL);
+
+        // scrollbar ranges
+        hbar.setMin(0);
+        vbar.setMin(0);
+        hbar.maxProperty().bind(content.totalWidthEstimateProperty());
+        vbar.maxProperty().bind(content.totalHeightEstimateProperty());
+
+        // scrollbar increments
+        setupUnitIncrement(hbar);
+        setupUnitIncrement(vbar);
+        hbar.blockIncrementProperty().bind(hbar.visibleAmountProperty());
+        vbar.blockIncrementProperty().bind(vbar.visibleAmountProperty());
+
+        // scrollbar positions
+        hbarValue = Var.doubleVar(hbar.valueProperty());
+        vbarValue = Var.doubleVar(vbar.valueProperty());
+        Bindings.bindBidirectional(
+                hbarValue,
+                content.estimatedScrollXProperty());
+        Bindings.bindBidirectional(
+                vbarValue,
+                content.estimatedScrollYProperty());
+
+        // scrollbar visibility
+        Val<Double> layoutWidth = Val.map(layoutBoundsProperty(), Bounds::getWidth);
+        Val<Double> layoutHeight = Val.map(layoutBoundsProperty(), Bounds::getHeight);
+        Val<Boolean> needsHBar0 = Val.combine(
+                content.totalWidthEstimateProperty(),
+                layoutWidth,
+                (cw, lw) -> cw > lw);
+        Val<Boolean> needsVBar0 = Val.combine(
+                content.totalHeightEstimateProperty(),
+                layoutHeight,
+                (ch, lh) -> ch > lh);
+        Val<Boolean> needsHBar = Val.combine(
+                needsHBar0,
+                needsVBar0,
+                content.totalWidthEstimateProperty(),
+                vbar.widthProperty(),
+                layoutWidth,
+                (needsH, needsV, cw, vbw, lw) -> needsH || needsV && cw + vbw.doubleValue() > lw);
+        Val<Boolean> needsVBar = Val.combine(
+                needsVBar0,
+                needsHBar0,
+                content.totalHeightEstimateProperty(),
+                hbar.heightProperty(),
+                layoutHeight,
+                (needsV, needsH, ch, hbh, lh) -> needsV || needsH && ch + hbh.doubleValue() > lh);
+        hbar.visibleProperty().bind(needsHBar);
+        vbar.visibleProperty().bind(needsVBar);
+
+        // request layout later, because if currently in layout, the request is ignored
+        hbar.visibleProperty().addListener(obs -> Platform.runLater(() -> requestLayout()));
+        vbar.visibleProperty().addListener(obs -> Platform.runLater(() -> requestLayout()));
+
+        getChildren().addListener((Observable obs) -> dispose());
```

Comment:
Ah... I think I get what's happening now.
If you look at the mouse, the scroll bar only barely appears. So I guess it is technically working.
![rich text demo_003](https://cloud.githubusercontent.com/assets/8413037/11102638/89c3239e-8872-11e5-82e1-63cb204bb6dd.jpg)